### PR TITLE
Cli service fix

### PR DESCRIPTION
**Pull request by orhovy**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -33,7 +33,7 @@ dependencies {
     implementation("com.beust:klaxon:5.5")
     implementation("com.google.code.gson:gson:2.8.8")
     implementation("org.json:json:20171018")
-
+    implementation("org.json:json:20171018")
```
Comment (yuvalyacoby) [inline build.gradle.kts:36]: isn't that duplicate?

File: src/main/kotlin/com/bridgecrew/activities/PostStartupActivity.kt [hunk 2]
```diff
@@ -1,23 +1,90 @@
...
+
 private val LOG = logger<PostStartupActivity>()
 
 class PostStartupActivity : StartupActivity {
 
     override fun runActivity(project: Project) {
         LOG.info("Startup activity starting")
         this.installCheckovOnStartup(project)
+        PipCheckovRunner.getPythonUserBasePath(project)
+        getGitRepoName(project)
```
Comment (yuvalyacoby) [inline src/main/kotlin/com/bridgecrew/activities/PostStartupActivity.kt:28]: why is it here? don't we only need this value for the checkov runs ?

File: src/main/kotlin/com/bridgecrew/services/CheckovScanService.kt [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+
+    private fun getGroupedResults(res: String, project: Project, relativeFilePath: String): Pair<ResourceToCheckovResultsList, Int> {
+        val listOfCheckovResults = getFailedChecksFromResultString(res)
+        return Pair(groupResultsByResource(listOfCheckovResults, project, relativeFilePath), listOfCheckovResults.size)
+    }
+
+    private fun prepareExecCommand(filePath: String, project: Project, apiToken: String, pluginVersion: String): ArrayList<String> {
+        val execCommand = selectedCheckovRunner!!.getExecCommand(filePath, apiToken, gitRepo, pluginVersion)
```
Comment (yuvalyacoby) [inline src/main/kotlin/com/bridgecrew/services/CheckovScanService.kt:142]: wouldn't that always send the default git repo name?

File: src/main/kotlin/com/bridgecrew/services/CliService.kt [hunk 4]
```diff
@@ -1,43 +1,70 @@
...
-            LOG.info(error)
-            throw Exception("Failed to run cli command")
-        }
-
-        LOG.info("Finished running command: $commandToPrint")
-
-        return result
+        val generalCommandLine = GeneralCommandLine(commands)
+        generalCommandLine.charset = Charset.forName("UTF-8")
+        generalCommandLine.setWorkDirectory(project.getBasePath())/////////////??????????
```
Comment (yuvalyacoby) [inline src/main/kotlin/com/bridgecrew/services/CliService.kt:34]: is the comment on purpose ?

File: src/main/kotlin/com/bridgecrew/services/CliService.kt [hunk 5]
```diff
@@ -1,43 +1,70 @@
...
+        val processHandler: ProcessHandler,
+        val function: (output: String, exitCode: Int, project: Project) -> Unit,
+    ) :
+        Task.Backgroundable(project, title,true) {
+        override fun run(indicator: ProgressIndicator) {
+            indicator.isIndeterminate = false
+            val output =
+                ScriptRunnerUtil.getProcessOutput(processHandler,
+                    ScriptRunnerUtil.STDOUT_OR_STDERR_OUTPUT_KEY_FILTER,
+                    720000000)
```
Comment (yuvalyacoby) [inline src/main/kotlin/com/bridgecrew/services/CliService.kt:60]: whats that number? perhaps it's better to use a variable that will indicate what it is

**Review by yuvalyacoby — APPROVED** [review]
Looks good, left minor comments

File: build.gradle.kts (hunk 1 above)
Comment (orhovy) [reply build.gradle.kts:36]: removed

File: src/main/kotlin/com/bridgecrew/activities/PostStartupActivity.kt (hunk 2 above)
Comment (orhovy) [reply src/main/kotlin/com/bridgecrew/activities/PostStartupActivity.kt:28]: As the plugin cannot run as a sequence of codes anymore and now is event driven.
On startup we run async all the necessary data for the runs, that way when a run is activated we can get the result.

File: src/main/kotlin/com/bridgecrew/services/CliService.kt (hunk 5 above)
Comment (orhovy) [reply src/main/kotlin/com/bridgecrew/services/CliService.kt:60]: A random number, changed it to be the default value of GeneralCommand object

File: src/main/kotlin/com/bridgecrew/services/CheckovScanService.kt (hunk 3 above)
Comment (orhovy) [reply src/main/kotlin/com/bridgecrew/services/CheckovScanService.kt:142]: No, as in startup we run getRepo and update this value in the gitUtils
`project.service<CheckovScanService>().gitRepo = result`

**Pull request merged**